src/lib.rs

```diff
@@ -168,8 +168,16 @@ impl Tokenizer {
             .map(|(string, token)| (token, string.chars().map(|ch| byte_decoder[&ch]).collect()))
             .collect();
 
-        let word_split =
-            Regex::new(r"'s|'t|'re|'ve|'m|'ll|'d|[\p{L}]+|[\p{N}]|[^\s\p{L}\p{N}]+").unwrap();
+        let word_split = Regex::new(
+            r"(?x)
+                # Special substrings - these each get encoded as a single marker token
```

Comment:
So one thing that I often remark on in PRs is that it's better to have refactoring and functional changes in separate commits. That makes it easier to review because you don't place the burden of mentally distinguishing these changes on the reviewer. So, while this is a fairly minor example of such a change, I'd like to suggest you make one commit to use the new formatting + comments approach for the old regex here, and then add the new cases in the second commit.